app/views/public/event_detail.py

```diff
@@ -84,7 +84,9 @@ def display_event_detail_home(identifier):
             sponsors[int(sponsor.level)] = [sponsor]
 
     fees = DataGetter.get_fee_settings_by_currency(event.payment_currency)
-    code = request.args.get("code")
+    acode = request.args.get("acode")
```

Comment:
Please refrain from using such names. Give a proper name like 
`access_code` and `discount_code`